docs/decisions/backend/0009-advanced-installer-design.md

```diff
@@ -0,0 +1,246 @@
+# Advanced Installer Design
+
+The 'Manual Mod Installer', also known as the 'Advanced Installer' is a mechanism for
+installing mods which do not come with an automatically recognizable file structure.
+
+This document lists the requirements for the given functionality, as well as some context.
+
+Associated UI Design:
+- https://www.figma.com/proto/Ntr9PttR5VrqpTRwwZHzLu/%F0%9F%93%B1-Mod-list?page-id=1826%3A5125&type=design&node-id=1826-5126&viewport=579%2C196%2C0.23&t=A6zTbYWd3oKZP1wF-1&scaling=min-zoom&starting-point-node-id=1826%3A5161
+
+## Context and Problem Statement
+
+There may be some situations where automated installation of a given mod might fail,
+or a user may want to install a mod manually.
+
+This can occur due to a variety of reasons; such as:
+- An oversight/bug with existing installer implementations.
+- An error on mod creator's part.
+- Unrecognized/non-standard packaging path.
+
+In the presence of these scenarios, the mod manager must still have a way to install any given mod.
+
+## Decision Drivers & Requirements
+
+- Certain mods may not come with a recognizable file structure.
+  - The Nexus Mods App might not know how to install them automatically.
+
+- Interoperability: The Nexus App must be able to somehow deploy ANY mod, regardless of its packaging format.
+  - Otherwise, if users end up having to deploy files manually, they may be discouraged from using the mod manager.
+
+- The Manual Installer must work with minimal game implementation code.
+  - It is considered the last line of defense; last resort.
+  - In other words, must not refuse to work in absence of support code for 'suggestions' etc.
+
+- UX: Must be Easy to Use
+
+  As part of our slogan named 'Make Modding Easy', it's important that the finished product is easy to use.
+
+  Namely, we have to consider that non-technical users may be using the app, and therefore we must make sure
+  to provide appropriate hints (suggestions) for them to deploy an unknown mod; even in presence of install
+  instructions from mod authors.
+
+- Must support arbitrary path mapping.
+
+  This system must support mapping any path from the source archive into any location in the target game directorie(s).
```

Comment:
To clarify, not just the game install directory, but also other game locations such as Appdata, MyGames etc